lib/services/cordova-migration-service.ts

```diff
@@ -134,9 +161,133 @@ export class CordovaMigrationService implements ICordovaMigrationService {
+				return;
+			}
+
+			this.$project.ensureAllPlatformAssets().wait();
+
+			if(this.$project.projectData.WPSdk && helpers.versionCompare(this.$project.projectData.WPSdk, "8.0") > 0 && helpers.versionCompare(newVersion, "3.7.0") < 0) {
+				let shouldUpdateWPSdk = this.$prompter.confirm(`You cannot build with the Windows Phone ${this.$project.projectData.WPSdk} SDK with the currently selected target version of Apache Cordova. Do you want to switch to Windows Phone 8.0 SDK?`).wait();
+				if(shouldUpdateWPSdk) {
+					this.onWPSdkVersionChanging("8.0").wait();
+					this.$project.projectData.WPSdk = "8.0";
+				} else {
+					this.$errors.failWithoutHelp("Unable to set %s as the target Apache Cordova version. Set the target Windows Phone SDK to 8.0 and try again.", newVersion);
+				}
+			}
+
+			let versionDisplayName = this.getDisplayNameForVersion(newVersion).wait();
+			this.$logger.info("Migrating to Cordova version %s", versionDisplayName);
+			let oldVersion = this.$project.projectData.FrameworkVersion;
+
+			_.each(this.$project.configurations, (configuration: string) => {
+				let oldPluginsList = this.$project.getProperty("CorePlugins", configuration);
+				let newPluginsList = this.migratePlugins(oldPluginsList, oldVersion, newVersion).wait();
+				this.$logger.trace("Migrated core plugins to: ", helpers.formatListOfNames(newPluginsList, "and"));
+				this.$project.setProperty("CorePlugins", newPluginsList, configuration);
+			});
+
+			let backedUpFiles: string[] = [],
+				backupSuffix = ".backup";
+			try {
+				_.each(this.$mobileHelper.platformNames, (platform) => {
+					this.$logger.trace("Replacing cordova.js file for %s platform ", platform);
+					let cordovaJsFileName = path.join(this.$project.getProjectDir().wait(), util.format("cordova.%s.js", platform).toLowerCase());
+					let cordovaJsSourceFilePath = this.$resources.buildCordovaJsFilePath(newVersion, platform);
+					this.$fs.copyFile(cordovaJsFileName, cordovaJsFileName + backupSuffix).wait();
+					backedUpFiles.push(cordovaJsFileName);
+					this.$fs.copyFile(cordovaJsSourceFilePath, cordovaJsFileName).wait();
+				});
+			} catch(error) {
+				_.each(backedUpFiles, file => {
+					this.$logger.trace("Reverting %s", file);
+					this.$fs.copyFile(file + backupSuffix, file).wait();
+				});
+				this.$errors.failWithoutHelp(error.message);
+			}
+			finally {
+				_.each(backedUpFiles, file => {
+					this.$fs.deleteFile(file + backupSuffix).wait();
+				});
+			}
+
+			this.$logger.info("Successfully migrated to version %s", versionDisplayName);
+		}).future<void>()();
+	}
+
+	public getSupportedPlugins(): IFuture<string[]> {
+		return (() => {
+			let version: string;
+			if(this.$project.projectData) {
+				version = this.$project.projectData.FrameworkVersion;
+			} else {
+				let selectedFramework = _.last(_.select(this.getSupportedFrameworks().wait(), (sv: Server.FrameworkVersion) => sv.DisplayName.indexOf(this.$projectConstants.EXPERIMENTAL_TAG) === -1));
```

Comment:
In terms of **lodash** `select` and `filter` are aliases. 
What this code actually does is get the latest non-experimental cordova version and call `pluginsForVersion` for that version, which I'm not entirely sure is correct